### PR TITLE
Support for aws_vpc support new attribute Ipv6CidrBlockNetworkBorderGroup

### LINKED ISSUE #15766: aws_vpc support new attribute Ipv6CidrBlockNetworkBorderGroup
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

1. Support the Ipv6CidrBlockNetworkBorderGroup attribute to provide IPv6 addresses to LocalZones and Wavelength Zones.

### New or Affected Resource(s)

* aws_vpc

### Potential Terraform Configuration

```hcl
ipv6_cidr_block_network_border_group = The name of the location from which we advertise the IPV6 CIDR block. Use this parameter to limit the address to this location. 
```

### References

* https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateVpc.html

### PR DESCRIPTION
**Pull request by NetDevAutomate**
aws_vpc support new attribute Ipv6CidrBlockNetworkBorderGroup

Closes #15766.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccVPC_assignGeneratedIPv6CIDRBlockWithBorder' ACCTEST_PARALLELISM=5 PKG=ec2

...
```

### CODE REVIEW

**Comment by NetDevAutomate** [comment]
$ make testacc TESTARGS='-run=TestAccVPC_' ACCTEST_PARALLELISM=5 PKG=ec2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 5 -run=TestAccVPC_ -timeout 180m
=== RUN   TestAccVPC_basic
=== PAUSE TestAccVPC_basic
=== RUN   TestAccVPC_disappears
=== PAUSE TestAccVPC_disappears
=== RUN   TestAccVPC_DefaultTags_providerOnly
=== PAUSE TestAccVPC_DefaultTags_providerOnly
=== RUN   TestAccVPC_DefaultTags_updateToProviderOnly
=== PAUSE TestAccVPC_DefaultTags_updateToProviderOnly
=== RUN   TestAccVPC_DefaultTags_updateToResourceOnly
=== PAUSE TestAccVPC_DefaultTags_updateToResourceOnly
=== RUN   TestAccVPC_DefaultTagsProviderAndResource_nonOverlappingTag
=== PAUSE TestAccVPC_DefaultTagsProviderAndResource_nonOverlappingTag
=== RUN   TestAccVPC_DefaultTagsProviderAndResource_overlappingTag
=== PAUSE TestAccVPC_DefaultTagsProviderAndResource_overlappingTag
=== RUN   TestAccVPC_DefaultTagsProviderAndResource_duplicateTag
=== PAUSE TestAccVPC_DefaultTagsProviderAndResource_duplicateTag
=== RUN   TestAccVPC_DynamicResourceTagsMergedWithLocals_ignoreChanges
=== PAUSE TestAccVPC_DynamicResourceTagsMergedWithLocals_ignoreChanges
=== RUN   TestAccVPC_DynamicResourceTags_ignoreChanges
=== PAUSE TestAccVPC_DynamicResourceTags_ignoreChanges
=== RUN   TestAccVPC_defaultAndIgnoreTags
=== PAUSE TestAccVPC_defaultAndIgnoreTags
=== RUN   TestAccVPC_ignoreTags
=== PAUSE TestAccVPC_ignoreTags
=== RUN   TestAccVPC_assignGeneratedIPv6CIDRBlock
=== PAUSE TestAccVPC_assignGeneratedIPv6CIDRBlock
=== RUN   TestAccVPC_assignGeneratedIPv6CIDRBlockWithBorder
=== PAUSE TestAccVPC_assignGeneratedIPv6CIDRBlockWithBorder
=== RUN   TestAccVPC_tenancy
=== PAUSE TestAccVPC_tenancy
=== RUN   TestAccVPC_IpamIpv4BasicNetmask
=== PAUSE TestAccVPC_IpamIpv4BasicNetmask
=== RUN   TestAccVPC_IpamIpv4BasicExplicitCidr
=== PAUSE TestAccVPC_IpamIpv4BasicExplicitCidr
=== RUN   TestAccVPC_tags
=== PAUSE TestAccVPC_tags
=== RUN   TestAccVPC_update
=== PAUSE TestAccVPC_update
=== RUN   TestAccVPC_bothDNSOptionsSet
=== PAUSE TestAccVPC_bothDNSOptionsSet
=== RUN   TestAccVPC_disabledDNSSupport
=== PAUSE TestAccVPC_disabledDNSSupport
=== RUN   TestAccVPC_classicLinkOptionSet
=== PAUSE TestAccVPC_classicLinkOptionSet
=== RUN   TestAccVPC_classicLinkDNSSupportOptionSet
=== PAUSE TestAccVPC_classicLinkDNSSupportOptionSet
=== CONT  TestAccVPC_basic
=== CONT  TestAccVPC_assignGeneratedIPv6CIDRBlock
=== CONT  TestAccVPC_classicLinkDNSSupportOptionSet
=== CONT  TestAccVPC_classicLinkOptionSet
=== CONT  TestAccVPC_disabledDNSSupport
=== CONT  TestAccVPC_bothDNSOptionsSet
--- PASS: TestAccVPC_basic (73.92s)
--- PASS: TestAccVPC_classicLinkDNSSupportOptionSet (74.03s)
=== CONT  TestAccVPC_update
--- PASS: TestAccVPC_classicLinkOptionSet (74.61s)
=== CONT  TestAccVPC_tags
=== CONT  TestAccVPC_IpamIpv4BasicExplicitCidr
--- PASS: TestAccVPC_disabledDNSSupport (78.13s)
=== CONT  TestAccVPC_IpamIpv4BasicExplicitCidr
    acctest.go:749: skipping tests; missing IAM service-linked role /aws-service-role/ipam.amazonaws.com. Please create the role and retry
--- SKIP: TestAccVPC_IpamIpv4BasicExplicitCidr (1.28s)
=== CONT  TestAccVPC_IpamIpv4BasicNetmask
    acctest.go:749: skipping tests; missing IAM service-linked role /aws-service-role/ipam.amazonaws.com. Please create the role and retry
--- SKIP: TestAccVPC_IpamIpv4BasicNetmask (0.35s)
=== CONT  TestAccVPC_tenancy
--- PASS: TestAccVPC_bothDNSOptionsSet (107.65s)
=== CONT  TestAccVPC_assignGeneratedIPv6CIDRBlockWithBorder
=== CONT  TestAccVPC_DefaultTagsProviderAndResource_overlappingTag
--- PASS: TestAccVPC_assignGeneratedIPv6CIDRBlock (228.05s)
=== CONT  TestAccVPC_ignoreTags
--- PASS: TestAccVPC_update (164.49s)
--- PASS: TestAccVPC_tags (213.31s)
=== CONT  TestAccVPC_defaultAndIgnoreTags
=== CONT  TestAccVPC_DynamicResourceTags_ignoreChanges
--- PASS: TestAccVPC_tenancy (211.25s)
--- PASS: TestAccVPC_ignoreTags (111.65s)
=== CONT  TestAccVPC_DynamicResourceTagsMergedWithLocals_ignoreChanges
--- PASS: TestAccVPC_assignGeneratedIPv6CIDRBlockWithBorder (171.18s)
=== CONT  TestAccVPC_DefaultTagsProviderAndResource_duplicateTag
--- PASS: TestAccVPC_DefaultTagsProviderAndResource_overlappingTag (126.96s)
=== CONT  TestAccVPC_DefaultTags_providerOnly
--- PASS: TestAccVPC_DefaultTagsProviderAndResource_duplicateTag (3.26s)
=== CONT  TestAccVPC_DefaultTagsProviderAndResource_nonOverlappingTag
--- PASS: TestAccVPC_DynamicResourceTags_ignoreChanges (91.17s)
=== CONT  TestAccVPC_DefaultTags_updateToResourceOnly
=== CONT  TestAccVPC_DefaultTags_updateToProviderOnly
--- PASS: TestAccVPC_defaultAndIgnoreTags (99.99s)
--- PASS: TestAccVPC_DynamicResourceTagsMergedWithLocals_ignoreChanges (87.40s)
=== CONT  TestAccVPC_disappears
--- PASS: TestAccVPC_DefaultTags_updateToResourceOnly (78.80s)
--- PASS: TestAccVPC_DefaultTags_updateToProviderOnly (75.21s)
--- PASS: TestAccVPC_disappears (33.12s)
--- PASS: TestAccVPC_DefaultTagsProviderAndResource_nonOverlappingTag (117.97s)
--- PASS: TestAccVPC_DefaultTags_providerOnly (119.41s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	481.978s

File: .changelog/22211.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
+resource/ec2/vpc: Add `ipv6_cidr_block_network_border_group` argument
```
Comment (drewmullen) [inline .changelog/22211.txt:2]: should be `resource/aws_vpc:`

File: website/docs/r/vpc.html.markdown [hunk 2]
```diff
@@ -102,6 +103,7 @@ In addition to all arguments above, the following attributes are exported:
 * `default_security_group_id` - The ID of the security group created by default on VPC creation
 * `default_route_table_id` - The ID of the route table created by default on VPC creation
 * `ipv6_association_id` - The association ID for the IPv6 CIDR block.
+* ipv6_cidr_block_network_border_group - The Network Border Group Zone name
```
Comment (drewmullen) [inline website/docs/r/vpc.html.markdown:106]: all attribute names in the docs should be in single backtick `. 

please mention if an attribute has a required attribute in order to be set (aka: `assign_generated_ipv6_cidr_block`)

File: internal/service/ec2/vpc_test.go [hunk 3]
```diff
@@ -593,6 +600,67 @@ func TestAccVPC_assignGeneratedIPv6CIDRBlock(t *testing.T) {
...
+					testAccCheckVpcCidr(&vpc, "10.1.0.0/16"),
+					resource.TestCheckResourceAttr(resourceName, "assign_generated_ipv6_cidr_block", "false"),
+					resource.TestCheckResourceAttr(resourceName, "cidr_block", "10.1.0.0/16"),
+					resource.TestCheckResourceAttr(resourceName, "ipv6_cidr_block_network_border_group", ""),
+					resource.TestCheckResourceAttr(resourceName, "ipv6_association_id", ""),
+					resource.TestCheckResourceAttr(resourceName, "ipv6_cidr_block", ""),
+				),
+			},
+			{
+				Config: testAccVpcConfigAssignGeneratedIpv6CidrBlockWithBorder(true, "us-west-2-lax-1"),
```
Comment (drewmullen) [inline internal/service/ec2/vpc_test.go:650]: IMO we can ignore this lint error // lintignore:AWSAT003

File: internal/service/ec2/vpc_test.go [hunk 4]
```diff
@@ -544,14 +545,20 @@ func TestAccVPC_assignGeneratedIPv6CIDRBlock(t *testing.T) {
 	var vpc ec2.Vpc
 	resourceName := "aws_vpc.test"
 
+	key := "AWS_DEFAULT_REGION"
+	regionId := os.Getenv(key)
+	if regionId == "" {
+		regionId = "us-east-1"
```
Comment (drewmullen) [inline internal/service/ec2/vpc_test.go:551]: is there a reason us-east-1 is hardcoded here?

**Review by drewmullen — CHANGES_REQUESTED** [review]

**Comment by NetDevAutomate** [comment]
@ewbankkit Many thanks for the support. If I may ask, why don't we ignore the .tflint.hcl' files?

**Comment by ewbankkit** [comment]
@NetDevAutomate The `.tflint.hcl` file that is checked in is configuring the various `tflint` executions run by GitHub actions.

**Comment by NetDevAutomate** [comment]
> @NetDevAutomate The `.tflint.hcl` file that is checked in is configuring the various `tflint` executions run by GitHub actions.

Thank you, that makes sense. Still lots to learn and as always very grateful for the support. Much appreciated.

**Comment by NetDevAutomate** [comment]
> As discussed elsewhere, I think you'll need to add `Computed: true` to the new argument as a default value of the current AWS Region is assigned. This can be verified via a now failing test of the `aws_vpc` data source:
> 
> ```
> === CONT  TestAccEC2VPCDataSource_ipv6Associated
>     vpc_data_source_test.go:92: Step 1/1 error: After applying this test step, the plan was not empty.
>         stdout:
>         
>         
>         Terraform used the selected providers to generate the following execution
>         plan. Resource actions are indicated with the following symbols:
>           ~ update in-place
>          <= read (data resources)
>         
>         Terraform will perform the following actions:
>         
>           # data.aws_vpc.by_id will be read during apply
>           # (config refers to values not yet known)
>          <= data "aws_vpc" "by_id"  {
>               ~ arn                     = "arn:aws:ec2:us-west-2:123456789012:vpc/vpc-03c7db19424697981" -> (known after apply)
>               ~ cidr_block              = "10.93.0.0/16" -> (known after apply)
>               ~ cidr_block_associations = [
>                   - {
>                       - association_id = "vpc-cidr-assoc-0fae851f23aefcf2c"
>                       - cidr_block     = "10.93.0.0/16"
>                       - state          = "associated"
>                     },
>                 ] -> (known after apply)
>               ~ default                 = false -> (known after apply)
>               ~ dhcp_options_id         = "dopt-11ac7e74" -> (known after apply)
>               ~ enable_dns_hostnames    = false -> (known after apply)
>               ~ enable_dns_support      = true -> (known after apply)
>                 id                      = "vpc-03c7db19424697981"
>               ~ instance_tenancy        = "default" -> (known after apply)
>               ~ ipv6_association_id     = "vpc-cidr-assoc-00359e176d59ccdbd" -> (known after apply)
>               ~ ipv6_cidr_block         = "2600:1f13:cf0:3b00::/56" -> (known after apply)
>               ~ main_route_table_id     = "rtb-00ac9a89b69467bdf" -> (known after apply)
>               ~ owner_id                = "123456789012" -> (known after apply)
>               ~ state                   = "available" -> (known after apply)
>               ~ tags                    = {
>                   - "Name" = "terraform-testacc-vpc-data-source-ipv6-associated-93"
>                 } -> (known after apply)
>             }
>         
>           # aws_vpc.test will be updated in-place
>           ~ resource "aws_vpc" "test" {
>                 id                                   = "vpc-03c7db19424697981"
>               - ipv6_cidr_block_network_border_group = "us-west-2" -> null
>                 tags                                 = {
>                     "Name" = "terraform-testacc-vpc-data-source-ipv6-associated-93"
>                 }
>                 # (17 unchanged attributes hidden)
>             }
>         
>         Plan: 0 to add, 1 to change, 0 to destroy.
> ```

Kit - fully agree, I got your comments through another mechanism and understand why. I've updated locally and running the tests to make sure they all pass. Assuming they do, I'll update the code.

**Comment by NetDevAutomate** [comment]
Modified as per @ewbankkit's advice.

$ make testacc TESTARGS='-run=TestAccVPC_' ACCTEST_PARALLELISM=5 PKG=ec2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 5 -run=TestAccVPC_ -timeout 180m
=== RUN   TestAccVPC_basic
=== PAUSE TestAccVPC_basic
=== RUN   TestAccVPC_disappears
=== PAUSE TestAccVPC_disappears
=== RUN   TestAccVPC_DefaultTags_providerOnly
=== PAUSE TestAccVPC_DefaultTags_providerOnly
=== RUN   TestAccVPC_DefaultTags_updateToProviderOnly
=== PAUSE TestAccVPC_DefaultTags_updateToProviderOnly
=== RUN   TestAccVPC_DefaultTags_updateToResourceOnly
=== PAUSE TestAccVPC_DefaultTags_updateToResourceOnly
=== RUN   TestAccVPC_DefaultTagsProviderAndResource_nonOverlappingTag
=== PAUSE TestAccVPC_DefaultTagsProviderAndResource_nonOverlappingTag
=== RUN   TestAccVPC_DefaultTagsProviderAndResource_overlappingTag
=== PAUSE TestAccVPC_DefaultTagsProviderAndResource_overlappingTag
=== RUN   TestAccVPC_DefaultTagsProviderAndResource_duplicateTag
=== PAUSE TestAccVPC_DefaultTagsProviderAndResource_duplicateTag
=== RUN   TestAccVPC_DynamicResourceTagsMergedWithLocals_ignoreChanges
=== PAUSE TestAccVPC_DynamicResourceTagsMergedWithLocals_ignoreChanges
=== RUN   TestAccVPC_DynamicResourceTags_ignoreChanges
=== PAUSE TestAccVPC_DynamicResourceTags_ignoreChanges
=== RUN   TestAccVPC_defaultAndIgnoreTags
=== PAUSE TestAccVPC_defaultAndIgnoreTags
=== RUN   TestAccVPC_ignoreTags
=== PAUSE TestAccVPC_ignoreTags
=== RUN   TestAccVPC_assignGeneratedIPv6CIDRBlock
=== PAUSE TestAccVPC_assignGeneratedIPv6CIDRBlock
=== RUN   TestAccVPC_assignGeneratedIPv6CIDRBlockWithBorder
=== PAUSE TestAccVPC_assignGeneratedIPv6CIDRBlockWithBorder
=== RUN   TestAccVPC_tenancy
=== PAUSE TestAccVPC_tenancy
=== RUN   TestAccVPC_IpamIpv4BasicNetmask
=== PAUSE TestAccVPC_IpamIpv4BasicNetmask
=== RUN   TestAccVPC_IpamIpv4BasicExplicitCidr
=== PAUSE TestAccVPC_IpamIpv4BasicExplicitCidr
=== RUN   TestAccVPC_tags
=== PAUSE TestAccVPC_tags
=== RUN   TestAccVPC_update
=== PAUSE TestAccVPC_update
=== RUN   TestAccVPC_bothDNSOptionsSet
=== PAUSE TestAccVPC_bothDNSOptionsSet
=== RUN   TestAccVPC_disabledDNSSupport
=== PAUSE TestAccVPC_disabledDNSSupport
=== RUN   TestAccVPC_classicLinkOptionSet
=== PAUSE TestAccVPC_classicLinkOptionSet
=== RUN   TestAccVPC_classicLinkDNSSupportOptionSet
=== PAUSE TestAccVPC_classicLinkDNSSupportOptionSet
=== CONT  TestAccVPC_basic
=== CONT  TestAccVPC_assignGeneratedIPv6CIDRBlock
=== CONT  TestAccVPC_update
=== CONT  TestAccVPC_tags
=== CONT  TestAccVPC_IpamIpv4BasicExplicitCidr
    acctest.go:749: skipping tests; missing IAM service-linked role /aws-service-role/ipam.amazonaws.com. Please create the role and retry
--- SKIP: TestAccVPC_IpamIpv4BasicExplicitCidr (2.75s)
=== CONT  TestAccVPC_IpamIpv4BasicNetmask
    acctest.go:749: skipping tests; missing IAM service-linked role /aws-service-role/ipam.amazonaws.com. Please create the role and retry
--- SKIP: TestAccVPC_IpamIpv4BasicNetmask (0.43s)
=== CONT  TestAccVPC_tenancy
--- PASS: TestAccVPC_basic (52.01s)
=== CONT  TestAccVPC_assignGeneratedIPv6CIDRBlockWithBorder
--- PASS: TestAccVPC_update (98.48s)
=== CONT  TestAccVPC_DefaultTagsProviderAndResource_overlappingTag
=== CONT  TestAccVPC_ignoreTags
--- PASS: TestAccVPC_tags (143.07s)
--- PASS: TestAccVPC_tenancy (141.99s)
=== CONT  TestAccVPC_defaultAndIgnoreTags
--- PASS: TestAccVPC_assignGeneratedIPv6CIDRBlock (145.17s)
=== CONT  TestAccVPC_DynamicResourceTags_ignoreChanges
--- PASS: TestAccVPC_assignGeneratedIPv6CIDRBlockWithBorder (140.71s)
=== CONT  TestAccVPC_DynamicResourceTagsMergedWithLocals_ignoreChanges
=== CONT  TestAccVPC_DefaultTagsProviderAndResource_duplicateTag
--- PASS: TestAccVPC_DefaultTagsProviderAndResource_overlappingTag (104.65s)
--- PASS: TestAccVPC_DefaultTagsProviderAndResource_duplicateTag (2.17s)
=== CONT  TestAccVPC_classicLinkOptionSet
=== CONT  TestAccVPC_classicLinkDNSSupportOptionSet
--- PASS: TestAccVPC_DynamicResourceTags_ignoreChanges (77.00s)
--- PASS: TestAccVPC_ignoreTags (89.77s)
=== CONT  TestAccVPC_disabledDNSSupport
=== CONT  TestAccVPC_bothDNSOptionsSet
--- PASS: TestAccVPC_defaultAndIgnoreTags (89.32s)
--- PASS: TestAccVPC_classicLinkOptionSet (47.28s)
=== CONT  TestAccVPC_DefaultTags_providerOnly
--- PASS: TestAccVPC_DynamicResourceTagsMergedWithLocals_ignoreChanges (74.06s)
=== CONT  TestAccVPC_DefaultTagsProviderAndResource_nonOverlappingTag
--- PASS: TestAccVPC_classicLinkDNSSupportOptionSet (48.68s)
=== CONT  TestAccVPC_DefaultTags_updateToResourceOnly
--- PASS: TestAccVPC_disabledDNSSupport (58.00s)
=== CONT  TestAccVPC_disappears
--- PASS: TestAccVPC_bothDNSOptionsSet (58.97s)
=== CONT  TestAccVPC_DefaultTags_updateToProviderOnly
--- PASS: TestAccVPC_disappears (29.01s)
--- PASS: TestAccVPC_DefaultTags_updateToResourceOnly (66.68s)
--- PASS: TestAccVPC_DefaultTags_providerOnly (97.72s)
--- PASS: TestAccVPC_DefaultTags_updateToProviderOnly (67.25s)
--- PASS: TestAccVPC_DefaultTagsProviderAndResource_nonOverlappingTag (95.36s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	365.862s

**Comment by ewbankkit** [comment]
@NetDevAutomate Looks like you forgot to push the commit(s) 😄.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```
--- PASS: TestAccEC2VPCDataSource_basic (35.11s)
--- PASS: TestAccVPC_DefaultTags_updateToResourceOnly (45.17s)
--- PASS: TestAccVPC_ignoreTags (55.44s)
--- PASS: TestAccVPC_tags (74.98s)
--- PASS: TestAccVPC_tenancy (68.15s)
--- PASS: TestAccVPC_assignGeneratedIPv6CIDRBlockWithBorder (69.39s)
--- PASS: TestAccVPC_classicLinkOptionSet (25.99s)
--- PASS: TestAccVPC_assignGeneratedIPv6CIDRBlock (69.65s)
--- PASS: TestAccVPC_classicLinkDNSSupportOptionSet (27.13s)
--- PASS: TestAccVPC_disappears (14.73s)
--- PASS: TestAccVPC_bothDNSOptionsSet (37.27s)
--- PASS: TestAccVPC_update (50.94s)
--- PASS: TestAccVPC_DynamicResourceTagsMergedWithLocals_ignoreChanges (41.61s)
--- PASS: TestAccVPC_disabledDNSSupport (36.23s)
--- PASS: TestAccVPC_defaultAndIgnoreTags (48.97s)
--- PASS: TestAccVPC_DynamicResourceTags_ignoreChanges (42.64s)
--- PASS: TestAccVPC_DefaultTags_updateToProviderOnly (37.75s)
--- PASS: TestAccEC2VPCDataSource_CIDRBlockAssociations_multiple (51.36s)
--- PASS: TestAccVPC_DefaultTagsProviderAndResource_overlappingTag (56.85s)
--- PASS: TestAccEC2VPCDataSource_ipv6Associated (29.07s)
--- PASS: TestAccVPC_DefaultTagsProviderAndResource_duplicateTag (1.07s)
--- PASS: TestAccVPC_basic (24.56s)
--- PASS: TestAccVPC_DefaultTags_providerOnly (55.44s)
--- PASS: TestAccVPC_DefaultTagsProviderAndResource_nonOverlappingTag (55.44s)
```

**Comment by ewbankkit** [comment]
@drewmullen I can merge once you're OK.

**Comment by drewmullen** [comment]
@ewbankkit LGTM and passed all BYOIP tests:

```bash
$ make testacc TESTS=TestAccVPCIpam_ByoipIPv6 PKG=ec2 ACCTEST_PARALLELISM=1
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 1 -run='TestAccVPCIpam_ByoipIPv6' -timeout 180m
=== RUN   TestAccVPCIpam_ByoipIPv6
=== PAUSE TestAccVPCIpam_ByoipIPv6
=== CONT  TestAccVPCIpam_ByoipIPv6
--- PASS: TestAccVPCIpam_ByoipIPv6 (2293.35s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	2296.647s
```

nice work @NetDevAutomate !

**Comment by NetDevAutomate** [comment]
Thank you, @ewbankkit and @drewmullen. As still early in my contributing days, your support has been invaluable and hugely appreciated. I now hope to get increasingly involved and take on bigger PRs.

Again, my most sincere thanks to you both.

**Pull request merged**